cli/types/tests/cypress-tests.ts

```diff
@@ -108,12 +108,79 @@ namespace CypressItsTests {
+  cy.wrap(a).invoke(options, 1) // $ExpectType Chainable<number>
+
+  const b: [() => string, (it: number) => number] = [returnsString, returnsNumberPlusOne]
+  cy.wrap(b).invoke(1, 82) // $ExpectType Chainable<number>
+  cy.wrap(b).invoke(1, 82, 4) // $ExpectError
+  cy.wrap(b).invoke(options, 1, 82) // $ExpectType Chainable<number>
+
+  const c: [() => string, (a: number, b: number) => number] = [returnsString, returnsSum]
+  cy.wrap(c).invoke(1, 82, 3) // $ExpectType Chainable<number>
+  cy.wrap(c).invoke(0, 82, 3) // $ExpectError
+  cy.wrap(c).invoke(options, 1, 82, 3) // $ExpectType Chainable<number>
+
+  const d: [(it: string) => string, (it: number) => number] = [returnsStringWithSuffix, returnsNumberPlusOne]
+  cy.wrap(d).invoke(0, 'a') // $ExpectType Chainable<string>
+  cy.wrap(d).invoke(1, 'a') // $ExpectError
+  cy.wrap(d).invoke(0, 23) // $ExpectError
+  cy.wrap(d).invoke(1, 23) // $ExpectType Chainable<number>
+  cy.wrap(d).invoke(options, 1, 23) // $ExpectType Chainable<number>
+
+  class TestClass {
+    foo(s: string, b: number): string | number
+    foo(s: string): string
+    foo(n: number): number
+    foo(o: string | number): string | number {
+      return typeof(o) === 'string' ? o + '1' : o + 1
+    }
+    bar(): number {
+      return 1
+    }
+  }
 
-  // unfortunately could not define more precise type
-  // in this case it should have been "number", but so far no luck
-  cy.wrap([returnsString, returnsNumber]).invoke(1) // $ExpectType Chainable<any>
+  const instance = new TestClass()
+  cy.wrap(instance).invoke('foo', 'a', 1) // $ExpectType Chainable<string | number>
+  cy.wrap(instance).invoke('foo', 'a') // $ExpectType Chainable<string>
+  cy.wrap(instance).invoke('foo', 1) // $ExpectType Chainable<number>
+  cy.wrap(instance).invoke('foo') // $ExpectError
+  cy.wrap(instance).invoke('bar') // $ExpectType Chainable<number>
+  cy.wrap(instance).invoke('bar', 1) // $ExpectError
+  cy.wrap(instance).invoke('foo', 1, 1) // $ExpectError
+
+  const obj = {
+    foo: (value: number, name = 'n') => `${ name } = ${ value + 1 }`,
+    bar: { baz: (a: number) => a + 1 },
+    bam: () => 1
+  }
+  cy.wrap(obj).invoke({ log: false }, 'foo', 1).should('equal', 'n = 2') // $ExpectType Chainable<string>
+  cy.wrap(obj).invoke('foo', 1).should('equal', 'n = 2') // $ExpectType Chainable<string>
+  cy.wrap(obj).invoke('foo', 5, 'b').should('equal', 'b = 6') // $ExpectType Chainable<string>
+  cy.wrap(obj).its('bar').invoke('baz', 2).should('equal', 3) // $ExpectType Chainable<number>
+  cy.wrap(obj).invoke('bam') // $ExpectType Chainable<number>
+  cy.wrap(obj).invoke('bam', 1) // $ExpectError
+  cy.wrap(obj).invoke('bar', 2) // $ExpectError
+  cy.wrap(obj).invoke('bar.baz', 2).should('equal', 3) // $ExpectError
+  cy.wrap(obj).invoke('foo', 5, 1) // $ExpectError
+  cy.wrap(obj).invoke('foo', 5, 'b', 1) // $ExpectError
+  cy.get('input').then(it => it[0]).invoke('checkValidity') // $ExpectError
```

Comment:
can you namepace the input element tests more, they get lost mixed like this with wrapped object tests. Also, this needs probable needs a few complementary tests like
```
// checkValidity is a valid method on jQuery object, right?
cy.get('input').invoke('checkValidity') // $ExpectType Chainable<boolean>
cy.get('input').its(0) // $ExpectType Chainable<HTMLInputElement>
  // but not on the underlying DOM element
  .invoke('checkValidity') // $ExpectError
```